tests/jballerina-unit-test/src/test/resources/test-src/query/query-action.bal

```diff
@@ -675,6 +675,48 @@ function testQueryActionWithDoClauseContainsCheck() {
     assertTrue(res is string && res == "str2");
 }
 
+function foo8() returns string{
+    string[] stringArray = ["Hello", " ", "World"];
+    error? unionResult = from var item in stringArray
+                            where item is string
+                            do {
+                                if item == "Hello" {
+                                    return item;
+                                }
+                            };
+    return "no match";
+}
+
+function foo9() returns string{
+    (string?)[] stringArray = ["Hello", " ", "World", ()];
+    error? unionResult = from var item in stringArray
+                            where item is string
+                            do {
+                                if item == "Hello" {
+                                    return item;
+                                }
+                            };
+    return "no match";
+}
+
+function foo10() returns string{
+    (string|int)[] stringArray = ["Hello", 2, "World", 4];
+    error? unionResult = from var item in stringArray
+                            where item is string
+                            do {
+                                if item == "Hello" {
+                                    return item;
```

Comment:
It seems this narrowed symbol is not considered. 
Can you please check below is related to the fix
```ballerina
function foo10() returns "Hello" {
    (string|int)[] stringArray = ["Hello", 2, "World", 4];
    error? unionResult = from var item in stringArray
                            where item is string
                            do {
                                if item == "Hello" {
                                    return item;
                                }
                            };
    return "Hello";
}
```

Reply:
I assume this is because we override the `item` symbol within the if check with the original symbol; which is the narrowed symbol at `where`

Reply:
The given example is working as expected with this change